src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,127 @@
+package storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import exceptions.DukeException;
+import tasklist.TaskList;
+import tasks.Deadlines;
+import tasks.DukeTask;
+import tasks.Events;
+import tasks.Todo;
+
+
+
+public class Storage {
+
+    private final Path dataFolder = Paths.get(System.getProperty("user.dir"), "data");
+    private final Path dukeTxt = Paths.get(System.getProperty("user.dir"), "data", "duke.txt");
+    private List<DukeTask> loadfile;
+
+    public Storage() {
+        this.loadfile = new ArrayList<>();
+    }
+
+    public List<DukeTask> load() throws DukeException {
+        return parseFile(this.getFileFromPath());
+    }
+
+    public File getFileFromPath() {
+        if (!Files.exists(this.dataFolder) || !Files.exists(this.dukeTxt)) {
+            System.out.println("Oops! You don't seem to have a load file!");
+            System.out.println("Creating one now!!\n");
+        }
+        try {
+            Files.createDirectory(this.dataFolder);
+            Files.createFile(this.dukeTxt);
+        } catch (IOException e) {
+            System.out.println("Nice! We found your load file!");
+            System.out.println("Loading...\n");
+        }
+
+        File loadData = this.dukeTxt.toFile();
+        return loadData;
+    }
+
+    /**
+     * Parses the txt Duke file
+     * @param data File to be parsed.
+     * @return A List of DukeTasks.
+     * @throws DukeException
```

Comment:
Please explain when the exception will be thrown.

Reply:
Noted!!